### PR TITLE
Self-messages

### PR DESCRIPTION
**Pull request by ghost**
Implement receiving self-messages sent from other Steam connections when there's a compatible BitlBee.

- Run time: Checks if the "self_messages" global setting exists.
- Build time: Sets OPT_SELFMESSAGE to 0 if OPT_SELFMESSAGE doesn't exist.

(Compatibility checking based on https://github.com/jgeboski/bitlbee-facebook/commit/0c8ce52982a9846508464b1ca21ecc629a764bb6)

### CODE REVIEW

**Comment by dequis** [comment]
Whoa what, that was unexpected.

File: steam/steam.c [hunk 1]
```diff
@@ -202,6 +206,20 @@ static void steam_user_msg(SteamData *sata, SteamUserMsg *msg, gint64 time)
...
+    case STEAM_USER_MSG_TYPE_MY_SAYTEXT:
+        if (set_find(&sata->ic->bee->set, "self_messages") == NULL)
+            return;
+
+        if (msg->type == STEAM_USER_MSG_TYPE_MY_EMOTE)
+            str = g_strconcat("/me ", msg->text, NULL);
+        else
+            str = g_strdup(msg->text);
+
+		imcb_buddy_msg(sata->ic, sid, str, OPT_SELFMESSAGE, time);
```
Comment (dequis) [inline steam/steam.c]: This line has tabs instead of spaces for indentation

Comment (edwardmuel) [inline steam/steam.c]: Didn't notice that. Fixed it. (I checked all other whitespace but apparently missed that line)

**Comment by chauffer** [comment]
For what it's worth, I've tried building bitlbee on the self-messages branch and this works fine for me. 
Emotes ("/me text") on Steam are also properly translated to IRC actions.

**Comment by dequis** [comment]
What's up with the commit author appearing as "nokobon" instead of "edwardmuel"?

**Comment by jgeboski** [comment]
Any reason you are making the source code executable?  I am assuming this is the result of your editor or filesystem.

This is pending bitlbee/bitlbee#49 getting merged.  Kudos for doing this, many will love you (#43, #63)!

**Comment by edwardmuel** [comment]
No, I didn't even know I made it executable. I cloned the repository on my laptop (running Mint) and did the editing over a Samba share using Notepad++, so my guess would be N++.

Would that change show up using git diff before committing? I don't see anything from just looking at the commits online.

**Comment by jgeboski** [comment]
If you look at the [patch](https://github.com/edwardmuel/bitlbee-steam/commit/b841247643d43981e71c66d0f21ee49a3e8aaa0b.patch), right after the `diff --git` line, you will see `old mode` and `new mode`.  I am also inclined to blame Samba, as it is converting between Windows and POSIX permissions.  Either way, the code should not be executable.

**Comment by edwardmuel** [comment]
I did a quick bit of searching... Apparently, it is both N++ and Samba's fault.

N++ sets the archive bit and then Samba converts that to the executable bit by default according to http://superuser.com/q/541270/strange-permission-changes-when-saving-file-on-a-samba-partition-from-a-windows :(

**Comment by chauffer** [comment]
Upstream selfmessage PR has been merged, time to merge this too?

**Comment by jgeboski** [comment]
Squashed and merged as ec3be01.

@edwardmuel you did not set a proper commit email. If you provide me an email, I'll update the patch before it is merged into `master`. This way it's easier to credit you for your work.

**Comment by chauffer** [comment]
Might as well merge this in master, they deleted their account.

**Comment by jgeboski** [comment]
Done.

**Pull request closed without merging**